google/cloud/security/scanner/audit/org_rules_engine.py

```diff
@@ -23,18 +23,90 @@
-from google.cloud.security.scanner.audit.base_rules_engine import BaseRuleBook
-from google.cloud.security.scanner.audit.base_rules_engine import BaseRulesEngine
+from google.cloud.security.scanner.audit import base_rules_engine
 from google.cloud.security.scanner.audit.errors import InvalidRulesSchemaError
-# pylint: enable=line-too-long
 
 
-class OrgRulesEngine(BaseRulesEngine):
+def _check_whitelist_members(rule_members=None, policy_members=None):
+    """Whitelist: Check that policy members ARE in rule members.
+
+    If a policy member is NOT found in the rule members, add it to
+    the violating members.
+
+    Args:
+        rule_members: A list of IamPolicyMembers allowed in the rule.
+        policy_members: A list of IamPolicyMembers in the policy.
+
+    Return:
+        A list of the violating members: policy members NOT found in
+        the whitelist (rule members).
+    """
+    violating_members = []
+    for policy_member in policy_members:
+        found_in_rules = False
+        for rule_member in rule_members:
+            found_in_rules = rule_member.matches(policy_member)
+            if found_in_rules:
+                break
+        if not found_in_rules:
+            violating_members.append(policy_member)
+    return violating_members
+
+def _check_blacklist_members(rule_members=None, policy_members=None):
+    """Blacklist: Check that policy members ARE NOT in rule members.
+
+    If a policy member is found in the rule members, add it to the
+    violating members.
+
+    Args:
+        rule_members: A list of IamPolicyMembers allowed in the rule.
+        policy_members: A list of IamPolicyMembers in the policy.
+
+    Return:
+        A list of the violating members: policy members found in
+        the blacklist (rule members).
+    """
+    violating_members = []
+    for policy_member in policy_members:
+        for rule_member in rule_members:
+            if rule_member.matches(policy_member):
+                violating_members.append(policy_member)
+                break
+    return violating_members
+
+def _check_required_members(rule_members=None, policy_members=None):
+    """Required: Check that rule members are in policy members.
+
+    If a required rule member is NOT found in the policy members, add
+    it to the violating members. Note that the check is different:
+    it's reversed from the whitelist/blacklist (policy as a subset of
+    rules vs rules as subset of policy).
+
+    Args:
+        rule_members: A list of IamPolicyMembers allowed in the rule.
+        policy_members: A list of IamPolicyMembers in the policy.
+
+    Return:
+        A list of the violating members: rule members not found in the
+        policy (required-whitelist).
+    """
+    violating_members = []
+    for rule_member in rule_members:
+        found_in_rules = False
+        for policy_member in policy_members:
+            found_in_rules = rule_member.matches(policy_member)
```

Comment:
You can drop the need for found_in_rules

``
if rule_member.matches(policy_member):
 break

violating_members.append(rule_member)
``